cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/scripts/ScriptEvaluator.java

```diff
@@ -0,0 +1,122 @@
+
+    private static final String TRUE = "true";
+    private static final String FALSE = "false";
+    private static String LINE_SEPARATOR = System.lineSeparator();
+    private static final String SYSTEM_PROPERTIES_MAP = "__sys_prop__";
+    private static final String GET_FUNCTION_DEFINITION =
+            "def get(key, default_value):" + LINE_SEPARATOR +
+                    "  value = globals().get(key)" + LINE_SEPARATOR +
+                    "  return default_value if value is None else value";
+    private static final String GET_SP_FUNCTION_DEFINITION =
+            "def get_sp(key, default_value=None):" + LINE_SEPARATOR +
+                    "  property_value = __sys_prop__.get(key)" + LINE_SEPARATOR +
+                    "  return default_value if property_value is None else property_value";
+
+    @Autowired
+    @Qualifier("evalInterpreter")
+    private PythonInterpreter interpreter;
+
+    public Serializable evalExpr(
+            String expr,
+            Map<String, ? extends Serializable> context,
+            Set<SystemProperty> systemProperties){
+        return evalExpr(expr, context, systemProperties, new HashSet<ScriptFunction>());
+    }
+
+    //we need this method to be synchronized so we will not have multiple scripts run in parallel on the same context
+    public synchronized Serializable evalExpr(
+            String expr,
+            Map<String, ? extends Serializable> context,
+            Set<SystemProperty> systemProperties,
+            Set<ScriptFunction> functionDependencies) {
+        try {
+            cleanInterpreter(interpreter);
+            prepareInterpreterContext(context);
+            addFunctionsToContext(systemProperties, functionDependencies);
+            return eval(interpreter, expr);
+        } catch (Exception exception) {
+            throw new RuntimeException(
+                    "Error in running script expression: '"
+                            + expr + "',\n\tException is: " + exception.getMessage(), exception);
+        }
+    }
+
+    private void addFunctionsToContext(Set<SystemProperty> systemProperties, Set<ScriptFunction> functionDependencies) {
+        String functions = "";
+        for (ScriptFunction function : functionDependencies) {
+            switch (function) {
+                case GET:
+                    functions += GET_FUNCTION_DEFINITION;
+                    functions = appendDelimiterBetweenFunctions(functions);
+                    break;
+                case GET_SYSTEM_PROPERTY:
+                    interpreter.set(SYSTEM_PROPERTIES_MAP, prepareSystemProperties(systemProperties));
+                    functions += GET_SP_FUNCTION_DEFINITION;
+                    functions = appendDelimiterBetweenFunctions(functions);
+                    break;
+            }
+        }
+
+        if (StringUtils.isNotEmpty(functions)) {
+            functions = trimScript(functions);
+            exec(interpreter, functions);
```

Comment:
exactly - that is how the interpreter knows about the functions with the current context